### DISCUSSION TITLE
Preserve code structure after replacement?

### DISCUSSION THREAD

**Discussion by etiennebacher** [ast-grep/ast-grep#1514] [Q&A]
[Playground](https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6InB5dGhvbiIsInF1ZXJ5IjoiIiwicmV3cml0ZSI6IiIsInN0cmljdG5lc3MiOiJzbWFydCIsInNlbGVjdG9yIjoic3RhcnRfdGFnIiwiY29uZmlnIjoicnVsZTpcbiAgcGF0dGVybjogZXhwZWN0X2VxdWFsKG5hbWVzKCRWQVIpLCAkVkFMVUVTKVxuICAgIFxuZml4OiBleHBlY3RfbmFtZWQoJFZBUiwgJFZBTFVFUykiLCJzb3VyY2UiOiJleHBlY3RfZXF1YWwobmFtZXMoeCksIGMoJ2EnLCAnYicpKVxuXG5leHBlY3RfZXF1YWwoXG4gIG5hbWVzKHgpLFxuICBjKCdhJywgJ2InKVxuKSJ9)

I am looking for a way to preserve code formatting after ast-grep replacements. For instance, if I have the following code:

```
expect_equal(names(x), c('a', 'b'))

expect_equal(
  names(x),
  c('a', 'b')
)
```
and the following rule:
```yaml
rule:
  pattern: expect_equal(names($VAR), $VALUES)
    
fix: expect_named($VAR, $VALUES)
```

then the output is this:

```
expect_named(x, c('a', 'b'))

expect_named(x, c('a', 'b'))
```

while I would like it to be this:

```
expect_named(x, c('a', 'b'))

expect_named(
  x,
  c('a', 'b')
)
```

Is there a way to do this that wouldn't require me to write the patterns and fixes with this indentation?

**Comment by HerringtonDarkholme** [comment]
ast-grep will not format the code. Using a linter/formatter along with it is a better choice.

**Comment by etiennebacher** [comment]
That's what I thought, thank you for confirming
